ui/app/AppLayouts/Profile/views/MessagingView.qml

```diff
@@ -136,44 +136,52 @@ SettingsContentBase {
             components: [
                 StatusSwitch {
                     id: showMessageLinksSwitch
-                    checked: false
-                    onCheckedChanged: {
+                    function switchOffPreviewableSites() {
+                        //update all models
+                        localAccountSensitiveSettings.displayChatImages = false
+                        for (let i = 0; i < previewableSites.count; i++) {
+                            let item = previewableSites.get(i)
+                            RootStore.updateWhitelistedUnfurlingSites(item.address, false)
+                        }
+                    }
+                    checked: previewableSites.anyWhitelisted || localAccountSensitiveSettings.displayChatImages
+                    onToggled: {
                         if (checked === false) {
-                            // Switch all the whitelists to false
-                            imageSwitch.checked = false
-                            for (let i = 0; i < sitesListView.count; i++) {
-                                let item = sitesListView.itemAt(i)
-                                item.whitelistSwitch.checked = false
-                            }
+                            switchOffPreviewableSites()
                         }
                     }
                 }
             ]
             onClicked: {
-                showMessageLinksSwitch.checked = !showMessageLinksSwitch.checked
+                showMessageLinksSwitch.toggle()
+                if (showMessageLinksSwitch.checked === false) {
+                    showMessageLinksSwitch.switchOffPreviewableSites()
+                }
             }
         }
 
-        function populatePreviewableSites() {
+        function buildPreviewablesSitesJSON() {
             let whitelistAsString = root.messagingStore.getLinkPreviewWhitelist()
             if(whitelistAsString == "")
                 return
-            let whitelist = JSON.parse(whitelistAsString)
+
             if (!localAccountSensitiveSettings.whitelistedUnfurlingSites) {
                 localAccountSensitiveSettings.whitelistedUnfurlingSites = {}
             }
-            previewableSites.clear()
-            var oneEntryIsActive = false
+
+            let anyWhitelisted = false
+            let whitelist = JSON.parse(whitelistAsString)
             whitelist.forEach(entry => {
```

Comment:
You could simplify this `forEach` with just Array.some()

Reply:
This forEach has two responsibilities. The first one is to retrieve a value for entry.isWhitelisted from another model(property used in the delegate). The other is to update anyWhitelisted boolean.
Array.some() would be useful to provide a value for anyWhitelisted, but we cannot use it to also compute entry.isWhitelisted.